packages/cip2/test/RoundRobinRandomImprove.test.ts

```diff
@@ -56,6 +56,26 @@ describe('RoundRobinRandomImprove', () => {
         });
         expect(results.selection.inputs.size).toBe(1);
       });
+      it('Recomputes fee after selecting an extra utxo due to change not meeting minimumCoinQuantity', async () => {
+        const utxo = new Set([
```

Comment:
Could include a 4th utxo to check that no additional non-essential inputs are included. E.g.
```typescript
const utxo = new Set([
  TxTestUtil.createUnspentTxOutput({ coins: 1_000_000n }),
  TxTestUtil.createUnspentTxOutput({ coins: 1_000_000n }),
  TxTestUtil.createUnspentTxOutput({ coins: 1_000_000n }),
  TxTestUtil.createUnspentTxOutput({ coins: 975_000_000n })
]);
```
Then you also wouldn't need to worry about the order utxos are considered so you could just use:
`const random = jest.fn().mockReturnValue(0);`

Reply:
Actually, changing the `random` would result in a slightly different selection process (2 utxos secluded at roundrobin instead of 1), although the result is the same.

Reply:
Added a 4th utxo and a comment explaining how the test works